### PR TITLE
UI: Updating time filters

### LINKED ISSUE #1338: Editable timefilters
Adding one or multiple time filters is easy, but they cannot be edited. This fact makes it difficult to quickly change the time range while exploring.

This feature request is to make the time filter chip clickable and editable.

### PR DESCRIPTION
**Pull request by binglot**
Fix #1338. Allow the users to update the time filters that they've created by clicking on them. Also added some UX improvements, such as closing the dropdown when the filter is added/updated, or auto-populating the end time input based on the start time.

### CODE REVIEW

File: timesketch/frontend/src/components/Sketch/TimeFilter.vue [hunk 1]
```diff
@@ -116,6 +117,9 @@ export default {
       this.startDateTime = startDateTimeMoment.format(dateTimeTemplate)
       this.endDateTime = endDateTimeMoment.format(dateTimeTemplate)
 
+      // Change focus to the 'endDateTime' input
+      event.target.parentNode.parentNode.parentNode.getElementsByClassName('input')[1].focus()
```
Comment (berggren) [inline timesketch/frontend/src/components/Sketch/TimeFilter.vue:121]: This is a little difficult to follow. If you need to get the DOM element you can use the Ref. See this for an example:
https://vuejs.org/v2/guide/components-edge-cases.html#Accessing-Child-Component-Instances-amp-Child-Elements

File: timesketch/frontend/src/components/Sketch/TimeFilter.vue [hunk 2]
```diff
@@ -130,6 +134,19 @@ export default {
       this.$emit('addChip', this.chip)
       this.startDateTime = ''
       this.endDateTime = ''
+
+      // Close the menu
+      this.$parent.$parent.isActive = false;
```
Comment (berggren) [inline timesketch/frontend/src/components/Sketch/TimeFilter.vue:139]: This is difficult to reason about. What prop is being mutated here? In what component. Can you emit an event and catch that in the calling component instead?

File: timesketch/frontend/src/components/Sketch/TimeFilter.vue [hunk 3]
```diff
@@ -130,6 +134,19 @@ export default {
...
+    },
+    update: function() {
+      if (!(this.startDateTime && this.endDateTime)) {
+        return
+      }
+      this.chip['value'] = this.startDateTime + ',' + this.endDateTime;
+      this.$emit('updateChip', this.chip)
+
+      // Close the menu
+      this.$parent.$parent.isActive = false;
```
Comment (berggren) [inline timesketch/frontend/src/components/Sketch/TimeFilter.vue:149]: Same here

**Review by berggren — CHANGES_REQUESTED** [review]

**Comment by berggren** [comment]
First round done, @binglot PTAL

**Comment by binglot** [comment]
@berggren - ready for the 2nd round of review :)

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**